src/core/services.ts

```diff
@@ -1,76 +1,88 @@
+        decodeURIComponent(
+          String.fromCharCode.apply(
+            null,
+            new Uint8Array(details.requestBody.raw[0].bytes) as any
           )
-        );
+        )
+      );
 
-        send({
-          type: 'ON_REQUEST',
-          payload: {
-            request: {
-              requestId: details.requestId,
-              query: queryDetails.query,
-              variables: queryDetails.variables
-            }
-          }
-        });
-      }
-    },
-    { urls: ['<all_urls>'] },
-    ['requestBody']
-  );
-
-  //@ts-ignore
-  chrome.webRequest.onCompleted.addListener(
-    function (details: any) {
       send({
-        type: 'ON_REQUEST_COMPLETE',
+        type: 'ON_REQUEST',
         payload: {
-          requestId: details.requestId,
-          statusCode: details.statusCode
-        }
+          request: {
+            requestId: details.requestId,
+            query: queryDetails.query,
+            variables: queryDetails.variables,
+          },
+        },
       });
-    },
-    { urls: ['<all_urls>'] }
-  );
+    }
+  }
 
-  //@ts-ignore
-  chrome.webRequest.onErrorOccurred.addListener(
-    function (details: any) {
-      send({
-        type: 'ON_REQUEST_ERROR',
-        payload: {
-          requestId: details.requestId,
-          statusCode: details.error === canceled_error ? 'canceled' : 'error'
-        }
-      });
-    },
-    { urls: ['<all_urls>'] }
-  );
+  function onCompleted(details: any) {
+    send({
+      type: 'ON_REQUEST_COMPLETE',
+      payload: {
+        requestId: details.requestId,
+        statusCode: details.statusCode,
+      },
+    });
+  }
 
-  //@ts-ignore
-  chrome.webRequest.onBeforeSendHeaders.addListener(
-    function (details: any) {
-      send({
-        type: 'ON_BEFORE_SEND_HEADERS',
-        payload: {
-          requestId: details.requestId,
-          requestHeaders: details.requestHeaders
-        }
-      });
-    },
-    { urls: ['<all_urls>'] },
-    ['requestHeaders']
-  );
+  function onErrorOccurred(details: any) {
+    const canceled_error = 'net::ERR_ABORTED';
+
+    send({
+      type: 'ON_REQUEST_ERROR',
+      payload: {
+        requestId: details.requestId,
+        statusCode: details.error === canceled_error ? 'canceled' : 'error',
+      },
+    });
+  }
+
+  function onBeforeSendHeaders(details: any) {
+    send({
+      type: 'ON_BEFORE_SEND_HEADERS',
+      payload: {
+        requestId: details.requestId,
+        requestHeaders: details.requestHeaders,
+      },
+    });
+  }
+
+  function registerAllEvents(urls: string[]) {
+    Object.values(chromeListeners).map((event) =>
+      event({
+        urls,
+        onBeforeRequest,
+        onCompleted,
+        onErrorOccurred,
+        onBeforeSendHeaders,
+      })
+    );
+  }
+
+  const initialUrls = getUrls();
+
+  if (initialUrls.length) {
+    registerAllEvents(initialUrls);
+  }
+
+  onReceive((e: any) => {
+    if (e.type !== 'START_CHROME_LISTENERS') return;
```

Comment:
It's called whenever a new URL is either added or removed, right? Perhaps renaming it to `UPDATE_CHROME_LISTENERS` to better reflect that fact it runs multiple times? What do you think?